pkg/apis/config/store.go

```diff
@@ -46,8 +47,10 @@ func FromContextOrDefaults(ctx context.Context) *Config {
 		return cfg
 	}
 	defaults, _ := NewDefaultsFromMap(map[string]string{})
+	featureFlags, _ := NewFeatureFlagsFromMap(map[string]string{})
 	return &Config{
-		Defaults: defaults,
+		Defaults:     defaults,
+		FeatureFlags: featureFlags,
```

Comment:
Could you clarify why we need `FeatureFlags` to be separated from `Defaults`?
AFAIU feature flags are a special type of defaults, that take a boolean value

Reply:
> Thanks a lot for working on this, I've been wondering why we used the configStore for pipelineruns and not for taskruns too.
> My main questions on this is whether we really need the extra module, or if we could just use the existing one and add extra flags to it?
> 

Thank you for the detailed review. Please let me know if the following sounds correct. I'd be happy to update the PR if needed to merge the two sets of options.

I have implemented the draft this way mainly because the `feature-flags` and `config-defaults` are different config maps. Modules using the config store are only minimally affected by this: they use config store the same way, i.e. by passing the context, just specify the flags under `FeatureFlags` "option group" as `Defaults` and `FeatureFlags` are both structs under the same `config` module.

If combining the two sets is preferred, please let me know and I will update the PR.

Reply:
I tend to like having it seperate :thinking: 
/cc @sbwsg

Reply:
Indeed, `feature-flags` is already a separate ConfigMap.  If we wanted to merge it with `config-defaults` then I'd prefer that be a discussion/PR unto itself rather than mix it in with a behavioral change such as this.  @afrittoli if you prefer to go that route do you want to pursue it at an upcoming WG?